File: Second year/Design and Programming Environments/TicketManagerCSharp/TicketManagerCSharp/utils/EventChangeEvent.cs
using TicketManagerCSharp.domain;

namespace TicketManagerCSharp.utils
{
    public enum ChangeEventType
    {
        ADD, UPDATE, DELETE
    }

    public class EventChangeEvent : IEvent
    {
        private ChangeEventType Type { get; set; }
        private Event NewData { get; set; }
        private Event OldData { get; set; }

        public EventChangeEvent(ChangeEventType type, Event newData, Event oldData)
        {
            Type = type;
            NewData = newData;
            OldData = oldData;
        }

        public EventChangeEvent(ChangeEventType type, Event newData)
        {
            Type = type;
            NewData = newData;
        }
    }
}

File: Second year/Design and Programming Environments/TicketManagerJava/src/main/java/Utils/EventChangeEvent.java
package Utils;

import Domain.Event;

public class EventChangeEvent implements EventUtils{
    private EventType type;
    private Event newData, oldData;


    public EventChangeEvent(EventType type, Event newData, Event oldData) {
        this.type = type;
        this.newData = newData;
        this.oldData = oldData;
    }

    public EventChangeEvent(EventType type,Event newData){
        this.type = type;
        this.newData = newData;
    }

    public EventType getType(){
        return type;
    }

    public Event getNewData(){
        return newData;
    }

    public Event getOldData(){
        return oldData;
    }
}
